chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,218 @@
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
+
+## Upgrade Steps
+
+> **Note:** One can upgrade/downgrade between [pre-ChainIndexer](#previous-indexing-system) and [with-ChainIndexer](#chainindexer-indexing-system) Lotus versions without conflict because they persist state to different directories and don't rely on each other. No backup is necessary (but extra backups don't hurt).  
+
+Upgrading to the new `ChainIndexer` involves these steps:
+
+1. **Stop the Lotus Node**
+   - Stop your Lotus node before starting the upgrade and migration process.
+2. **Update Configuration**
+   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the [`ChainIndexer Config` section above](#chainindexer-config).
+3. **Restart Lotus Node**
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately in the `${LOTUS_PATH}/chainindex` directory.
+   - **However, it will not automatically index any historical chain state (i.e., any previously existing chain state). To perform backfilling, please see the [`Backfill` section below](#backfill).**
+
+> **Note:** It's recommended to keep the [pre-ChainIndexer](#previous-indexing-system) state directory around until you've confirmed you don't need to [downgrade](#downgrade).  After sustained successful operations after the upgrade, the [pre-ChainIndexer](#previous-indexing-system) state directory can be removed to reclaim disk space.  
+
+## Backfill
+There is no migration from [pre-ChainIndexer indices](#previous-indexing-system) to the [ChainIndex](#chainindexer-indexing-system) because the [pre-ChainIndexer indices](#previous-indexing-system) are not fully reliable.  Instead one needs to index historical chain state (i.e., backfill) using one of the following tools.
+
+> **Note:** The ChainIndex will consume ~10GB of storage per month of tipsets (e.g., ~86400 epochs).  Ensure you have sufficient disk space before proceeding.
+
+### `ChainValidateIndex` JSON RPC API
+
+Please refer to the [Lotus API documentation](https://github.com/filecoin-project/lotus/blob/master/documentation/en/api-v1-unstable-methods.md) for detailed documentation of the `ChainValidateIndex` JSON RPC API.
+
+The `ChainValidateIndex` JSON RPC API serves a dual purpose: it validates/diagnoses the integrity of the index at a specific epoch (i.e., it ensures consistency between indexed data and actual chain state), while also providing the option to backfill the `ChainIndexer` if it does not have data for the specified epoch. 
+
+The `ChainValidateIndex` RPC API is available for use once the Lotus daemon has started with `ChainIndexer` [enabled](#enablement). 
+
+Here are some examples of how to use the `ChainValidateIndex` JSON RPC API for validating/ backfilling the index:
```

Comment:
```suggestion
Here are some examples of how to use the `ChainValidateIndex` API manually with `curl` for validating/ backfilling the index (note there is a `lotus-shed` form of this for ease of use, see below):
```

how about we not scare them with curl up front? "you mean I need to curl each epoch I want to backfill?????"

Reply:
I also wonder if this is just too much detail for users, do they care about this? This whole manual `curl` section could even be an appendix if someone wanted to built out their own tooling, but really, who's going to do this? Can they not _trust us_?

Reply:
Good callout.  I'll move it down / out of view.

Reply:
Will handle in https://github.com/filecoin-project/lotus/pull/12450